### PR TITLE
[Reader] Fix drag and drop

### PR DESCRIPTION
**Pull request by NickHeiner**
Connect #2508.

To test, see the ticket AC.

drag and drop

### CODE REVIEW

File: client/app/components/Pdf.jsx [hunk 1]
```diff
@@ -752,18 +753,40 @@ export class Pdf extends React.PureComponent {
...
+
+        return;
+      }
+    } catch (err) {
+      if (err instanceof SyntaxError) {
+        logInvalidDragAndDrop();
+
+        return;
+      }
+      throw err;
```
Comment (tejans24) [inline client/app/components/Pdf.jsx]: Any particular reason for throwing this error again? Should we handle it a bit more gracefully?

**Comment by NickHeiner** [comment]
> It also looks like there is a lint issue, thus the tests failing.

Sorry about that! Fixed.

File: client/app/components/Pdf.jsx (hunk 1 above)
Comment (NickHeiner) [inline client/app/components/Pdf.jsx]: Yeah, this is a common pattern. There's one type of error that we know could occur in this situation. When that error happens, we gracefully handle it. However, it's possible that other errors could occur. We don't have a graceful way to handle them, so we just continue to propagate them.

You see something similar in Java where you have something like:

```java
catch (IOException e) {
  // ...
}
catch (ArgumentException e) {
  // ...
}
```

In Java, we're able to explicitly say, "I know how to handle an `IOException` or an `ArgumentException`, but I don't know about anything else, and it should not be caught". In JS, we're doing that same thing manually.

Comment (tejans24) [inline client/app/components/Pdf.jsx]: Right, that makes sense. I failed to notice the return in the if statement of handling the SyntaxError last night. 😑

**Pull request merged**